### PR TITLE
[xpu] pir conv2d add pass

### PR DESCRIPTION
**Pull request by gitliuyf**
### PR Category
Performance Optimization

### PR Types
New features

### Description
pir conv2d add pass

### CODE REVIEW

File: paddle/fluid/pir/transforms/xpu/conv2d_add_fuse_pass.cc [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+#include "paddle/pir/include/pass/pass.h"
+#include "paddle/pir/include/pass/pass_registry.h"
+#include "paddle/pir/include/pattern_rewrite/pattern_match.h"
+
+namespace {
+
+class Conv2dAddXpuFusePattern : public paddle::drr::DrrPatternBase {
+ private:
+  bool enable_int8_;
+  std::map<std::string, int> quant_post_type_;
```
Comment (yuanlehome) [inline paddle/fluid/pir/transforms/xpu/conv2d_add_fuse_pass.cc]:
```suggestion
  const std::map<std::string, int>& quant_post_type_;
```

File: paddle/fluid/pir/transforms/xpu/conv2d_add_fuse_pass.cc [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+  std::reference_wrapper<std::optional<pir::detail::PassExecutionState>>
+      pass_state_;
+
+ public:
+  explicit Conv2dAddXpuFusePattern(
+      bool enable_int8,
+      std::map<std::string, int> quant_post_type,
```
Comment (yuanlehome) [inline paddle/fluid/pir/transforms/xpu/conv2d_add_fuse_pass.cc]:
```suggestion
      const std::map<std::string, int>& quant_post_type,
```

File: paddle/pir/src/pass/pass.cc [hunk 3]
```diff
@@ -11,13 +11,14 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+#include "glog/logging.h"
```
Comment (yuanlehome) [inline paddle/pir/src/pass/pass.cc]: delete

File: paddle/pir/src/pass/pass.cc [hunk 4]
```diff
@@ -270,7 +270,9 @@ void PassInstrumentor::RunBeforeAnalysis(const std::string& name,
 void PassInstrumentor::RunAfterAnalysis(const std::string& name,
                                         TypeId id,
                                         Operation* op) {
+  VLOG(6) << "kyf -- before check num_regions";
```
Comment (yuanlehome) [inline paddle/pir/src/pass/pass.cc]: delete

File: paddle/pir/src/pass/pass.cc [hunk 5]
```diff
@@ -270,7 +270,9 @@ void PassInstrumentor::RunBeforeAnalysis(const std::string& name,
...
   if (op->num_regions() == 0) return;
+  VLOG(6) << "kyf -- after checknum_regions";
```
Comment (yuanlehome) [inline paddle/pir/src/pass/pass.cc]: delete

**Review by yuanlehome — APPROVED** [review]

**Review by hong19860320 — APPROVED** [review]
LGTM

**Pull request merged**